lib/ansible/modules/net_tools/basics/uri.py

```diff
@@ -479,6 +497,57 @@ def form_urlencoded(body):
     return body
 
 
+def form_multipart(body, boundary):
+    ''' Convert data into multipart/form-data '''
+    if isinstance(body, string_types):
+        return body
+
+    if isinstance(body, Mapping):
+        result = []
+        # Convert mapping into multipart chunks
+        for field, value in list(body.items()):
+            # result.extend(('--%s' % boundary))
+            if isinstance(value, string_types):
+                result.extend((
+                    '--%s' % boundary,
+                    'Content-Disposition: form-data; name="%s"' % field,
+                    '',
+                    str(value)
+                ))
+            if isinstance(value, Mapping):
+                # filename required
+                if 'filename' not in value:
+                    raise LookupError('required key "filename" missing from "%s" element' % field)
+                else:
+                    filename = os.path.basename(value['filename'])
+                # set or guess mimetype
+                if 'mimetype' in value:
+                    mimetype = value['mimetype']
+                else:
+                    mimetype = mimetypes.guess_type(filename)[0] or 'application/octet-stream'
+                # get the content or read the file from disk
+                if 'content' in value:
+                    data = value['content']
+                else:
+                    if os.path.exists(value['filename']):
+                        with open(value['filename'], 'rb') as data_file:
+                            data = data_file.read()
+                    else:
+                        raise IOError('unable to read file "%s"' % value['filename'])
+                result.extend((
+                    '--%s' % boundary,
+                    'Content-Disposition: form-data; name="%s"; filename="%s"' % (field, filename),
+                    'Content-Type: %s' % mimetype,
+                    '',
+                    to_text(data)
+                ))
+        result.extend((
+            '--%s--' % boundary,
+            ''
+        ))
+        return '\r\n'.join(result)
```

Comment:
```suggestion
        return b'\r\n'.join(to_bytes(line, nonstring='passthru') for line in result)
```
(and import `to_bytes`)